### PR TITLE
feat(EC-202): Proceed if Chains could not sign

### PR DESCRIPTION
**Pull request by lcarva**
This commit changes the controller to no longer wait for the PipelineRun to have the annotation `chains.tekton.dev/signed` be set to the value of `"true"`. Instead, it now only waits for the annotation to be set regardless of its value. This indicates the Chains controller is done processing the PipelineRun and the integration service should move on.

This prevents Chains failures from causing integration tests to never run. Instead, let the integration tests run, and verification tools, e.g. Enterprise Contract, report on unsigned artifacts.

## Maintainers will complete the following section

- [x] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [x] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [x] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

**Comment by lcarva** [comment]
When reviewing this PR, please help determine if you agree that the modified behavior is indeed better than the previous one.

**Review by dirgim — APPROVED** [review]
LGTM

File: tekton/predicates_test.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ var _ = Describe("Predicates", func() {
 			Expect(instance.Update(contextEvent)).To(BeFalse())
 		})
 
-		It("should return false when an updated event is received for a succeeded PipelineRun and not signed", func() {
+		It("should return true when an updated event is received for a succeeded PipelineRun and not signed", func() {
```
Comment (MartinBasti) [inline tekton/predicates_test.go:117]: can we rephrase it to something like `when an updated event is received for a succeeded PipelineRun with failed chains signing` ?

**Comment by MartinBasti** [comment]
Please update also description in diagram here:
https://github.com/redhat-appstudio/integration-service/blob/d7620c9eb233d1c406e721513ad9d1089cd6009b/docs/build_pipeline_controller.md?plain=1#L13

File: tekton/utils.go [hunk 2]
```diff
@@ -107,11 +107,19 @@ func hasPipelineRunStateChangedToDeleting(objectOld, objectNew client.Object) bo
...
+func isChainsDoneWithPipelineRun(objectNew client.Object) bool {
 	if newPipelineRun, ok := objectNew.(*tektonv1beta1.PipelineRun); ok {
-		return metadata.HasAnnotationWithValue(newPipelineRun, PipelineRunChainsSignedAnnotation, "true")
+		// If the annotation value is set "true" it means Chains was able to sign. If it is set to
+		// "failed", it means something prevented Chains from signing, e.g. insufficient access. In
+		// either case, we want to proceed processing the PipelineRun instead of waiting
+		// indefinitely for a condition that may never happen, e.g. "failed" -> "true". Let
+		// downstream verification processes, e.g. Enterprise Contract, deal with the Chains
+		// failure.
+		return metadata.HasAnnotation(newPipelineRun, PipelineRunChainsSignedAnnotation)
```
Comment (dheerajodha) [inline tekton/utils.go:122]: just to confirm: if we go by this approach, then we would only be checking if this annotation is present within the PLR even though it: 1) has no value, 2) is set to "true", or 3) is set to "failed".

Is it possible that this annotation could at some point be empty (or has no value) and cause this bug ([RHTAPBUGS-214](https://issues.redhat.com/browse/RHTAPBUGS-214)) again? Another part of me says it won't hit that bug because `Missing pipelinerun attestation` means _absence_ of this annotation.

Comment (lcarva) [reply tekton/utils.go:122]: Looking at the Chains code base does, the Chains controller does not set that annotation to an empty value. So this would only happen if something else does that. (But if there's something else playing with that annotation we're already in trouble.)

You may see `Missing pipelinerun attestation` again if Chains does fail to process the PipelineRun, but not because it hasn't had a chance to do so. This is the correct behavior IMO.

File: tekton/predicates_test.go (hunk 1 above)
Comment (lcarva) [reply tekton/predicates_test.go]: Done.

**Comment by lcarva** [comment]
> Please update also description in diagram

Done!

**Review by MartinBasti — APPROVED** [review]

File: tekton/utils.go (hunk 2 above)
Comment (dheerajodha) [reply tekton/utils.go:122]: That makes sense, thanks for the explanation!

**Review by dheerajodha — APPROVED** [review]
LGTM
